applications/DEM_application/custom_elements/rigid_body_element.cpp

```diff
@@ -1,100 +1,347 @@
-//
-//   Project Name:        Kratos
-//   Last Modified by:    $Author: Salva $
-//   Date:                $Date: 2017-01-09 16:07:33 $
-//   Revision:            $Revision: 1.1.1.1 $
-//
+// Created by: Salva Latorre, latorre@cimne.upc.edu
 
 // System includes
-#include <string>
-#include <iostream>
 #include <stdlib.h>
 
-// External includes
-
 // Project includes
 #include "rigid_body_element.h"
 #include "custom_utilities/GeometryFunctions.h"
 #include "custom_utilities/AuxiliaryFunctions.h"
 #include "DEM_application_variables.h"
 #include "includes/kratos_flags.h"
 #include "includes/variables.h"
-#include "custom_elements/spheric_continuum_particle.h"
+#include "geometries/point_3d.h"
 
 namespace Kratos {
 
-    RigidBodyElement::RigidBodyElement() : Element() {}
+    RigidBodyElement3D::RigidBodyElement3D() : Element() {
+    }
             
-    RigidBodyElement::RigidBodyElement(IndexType NewId, GeometryType::Pointer pGeometry)
+    RigidBodyElement3D::RigidBodyElement3D(IndexType NewId, GeometryType::Pointer pGeometry)
     : Element(NewId, pGeometry) {
         mpIntegrationScheme = NULL;
     }
       
-    RigidBodyElement::RigidBodyElement(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties)
+    RigidBodyElement3D::RigidBodyElement3D(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties)
     : Element(NewId, pGeometry, pProperties) {
         mpIntegrationScheme = NULL;
     }
       
-    RigidBodyElement::RigidBodyElement(IndexType NewId, NodesArrayType const& ThisNodes)
+    RigidBodyElement3D::RigidBodyElement3D(IndexType NewId, NodesArrayType const& ThisNodes)
     : Element(NewId, ThisNodes) {
         mpIntegrationScheme = NULL;
     }
     
-    Element::Pointer RigidBodyElement::Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const {
-        return Element::Pointer(new RigidBodyElement(NewId, GetGeometry().Create(ThisNodes), pProperties));
+    Element::Pointer RigidBodyElement3D::Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const {
+        return Element::Pointer(new RigidBodyElement3D(NewId, GetGeometry().Create(ThisNodes), pProperties));
     }      
     
     // Destructor
-    RigidBodyElement::~RigidBodyElement() {
-
+    RigidBodyElement3D::~RigidBodyElement3D() {
+    
+        mListOfCoordinates.clear();  
+        if (mpIntegrationScheme != NULL) delete mpIntegrationScheme;
+        // Destroy triangles?
```

Comment:
Watch out with this! You should not have the member variable mpIntegrationScheme both in RigidBodyElement3D AND in Cluster3D.